pkg/crosscluster/logical/replication_statements.go

```diff
@@ -254,8 +254,11 @@ func newDeleteStatement(
 			TableID: int64(table.GetID()),
 			As:      tree.AliasClause{Alias: "replication_target"},
 		},
-		Where:     &tree.Where{Type: tree.AstWhere, Expr: whereClause},
-		Returning: tree.AbsentReturningClause,
+		Where: &tree.Where{Type: tree.AstWhere, Expr: whereClause},
```

Comment:
@michae2 This feels like a bit of a hack. Is there a different way we should ensure delete is provided with every column in the previous row?

Reply:
This will require teaching the optimizer to not prune those columns when `OriginTimestampForLogicalDataReplication` is set. I have to do something similar for DeleteSwap in https://github.com/cockroachdb/cockroach/pull/145019/commits/8cad0448db7f320dff8ba70372db54ff21041fc7#diff-bc4853fa74b565a5f0dd097ad6b6654d4d2033b9cf81e99ebee6c5b889e01fc1 so I don't mind doing this, but it might be a few days. Let's leave the hack in place for now with a TODO(michae2) and I will try to remove it when I add the DeleteSwap rules to the optimizer.

Reply:
(If we're trying to backport this, probably best to backport it with the hack in place.)